cluster/plan.go

```diff
@@ -171,6 +171,11 @@ func (c *Cluster) BuildKubeAPIProcess(prefixPath string) v3.Process {
 	if c.Authorization.Mode == services.RBACAuthorizationMode {
 		CommandArgs["authorization-mode"] = "Node,RBAC"
 	}
+
+	if len(host.InternalAddress) > 0 {
```

Comment:
I think becuse of https://github.com/rancher/rke/blob/22dc19205f794f19dc156d16695ff86c07a501e6/cluster/defaults.go#L76-L78 there could be hostname in InternalAddress causing apiserver to fail to start - it requires IP. It happens when you do not specify internal IP for node and specified address is hostname.

Reply:
If InternalAddress meant to specify IP, there shouldn't be the hostname. 

see https://github.com/rancher/rke/blob/master/cluster/plan.go#L94 and https://github.com/rancher/rke/blob/master/k8s/node.go#L20

Reply:
@galal-hussein seems reasonable. applied. thanks.

Reply:
I just found adding the condition `host.InternalAddress != host.Address` can't solve my problem. In my case, the address is the same as the internal address, which is the ip of the private network interface, i want to specify the advertise-address arg to it since my public network ip is not accessible  by other nodes due to network policy.

Reply:
@willmao okay then, where do you put your accessable IP?

Reply:
@code0x9 the internal ip address provided by my cloud provider can be accessible by all nodes in the cluster, i think the advertise address should be set to the internal address once the `internal_address ` field is set and the format is ipv4 address(escape the address fqdn case)

sorry for the late reply!

Reply:
@willmao right. changed check logic from ```host.InternalAddress != host.Address``` to ```net.ParseIP(host.InternalAddress) != nil```, using golang builtin net package. thanks.

Reply:
@code0x9 my pleasure! hoping this pr could be merged before our kubernetes goes to production three month later...